src/libsync/networkjobs.cpp

```diff
@@ -834,52 +766,84 @@ bool DetermineAuthTypeJob::finished()
     return true;
 }
 
-SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, QObject *parent)
-    : AbstractNetworkJob(account, QString(), parent)
+SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, const QString &path, const QByteArray &verb, const QNetworkRequest &req, QObject *parent)
+    : AbstractNetworkJob(account, path, parent)
+    , _verb(verb)
+    , _request(req)
 {
 }
 
-void SimpleNetworkJob::start()
+SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, const QString &path, const QByteArray &verb, const UrlQuery &arguments, const QNetworkRequest &req, QObject *parent)
+    : SimpleNetworkJob(account, path, verb, req, parent)
 {
-    sendRequest(_simpleVerb, _simpleUrl, _simpleRequest, _simpleBody);
-    AbstractNetworkJob::start();
+    if (verb == QByteArrayLiteral("GET")) {
```

Comment:
I could imagine that you want to also check for `HEAD` requests there. Not sure if we use any, but we could in theory.